jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/log/RuntimeLogPresenter.java

```diff
@@ -53,60 +65,117 @@ public IsWidget getWidget() {
+                    (NODE_START.equals(logSummary.getNodeType()) && !logSummary.isCompleted())) {
+                agent = constants.Human();
+            }
+            return format(TECH_LOG_FORMAT,
+                          DateUtils.getDateTimeStr(logSummary.getDate()),
+                          logSummary.getNodeName(),
+                          logSummary.getNodeType(),
+                          (logSummary.isCompleted() ? " " + constants.Completed() : ""),
+                          agent
+            );
         } else {
-            processRuntimeDataService.call(new RemoteCallback<List<RuntimeLogSummary>>() {
-                @Override
-                public void callback(List<RuntimeLogSummary> logs) {
-                    final List<String> logsLine = new ArrayList<String>();
-                    if (logOrder == LogOrder.DESC) {
-                        Collections.reverse(logs);
-                    }
-
-                    for (RuntimeLogSummary rls : logs) {
-                        logsLine.add(rls.getTime() + ": " + rls.getLogLine());
-                    }
-
-                    view.setLogs(logsLine);
-                }
-            }).getBusinessLogs(currentServerTemplateId,
-                               currentDeploymentId,
-                               currentProcessName,
-                               currentProcessInstanceId);
+            String prettyTime = moment(DateTimeFormat.getFormat("yyyy-MM-dd hh:mm:ss a")
+                                               .format(logSummary.getDate()),
+                                       "YYYY-MM-DD hh:mm:ss a").fromNow();
+            if (NODE_HUMAN_TASK.equals(logSummary.getNodeType())) {
+                return format(BUSINESS_LOG_FORMAT,
+                              prettyTime,
+                              constants.Task(),
+                              logSummary.getNodeName(),
+                              (logSummary.isCompleted() ? constants.WasCompleted() : constants.WasCreated()));
+            } else if (NODE_START.equals(logSummary.getNodeType()) && !logSummary.isCompleted()) {
+                return format(BUSINESS_LOG_FORMAT,
+                              prettyTime,
+                              constants.Process(),
+                              getCurrentProcessName(),
+                              constants.WasCreated());
+            } else if (NODE_END.equals(logSummary.getNodeType()) && logSummary.isCompleted()) {
+                return format(BUSINESS_LOG_FORMAT,
+                              prettyTime,
+                              constants.Process(),
+                              getCurrentProcessName(),
+                              constants.WasCompleted());
+            }
+        }
+        return null;
+    }
+
+    private String format(String pattern,
+                          Object... params) {
+        String formattedText = pattern;
```

Comment:
@nmirasch true, String.format isn't available, you could give a go to GWT own templating then: http://www.gwtproject.org/doc/latest/DevGuideSecuritySafeHtml.html#SafeHtmlTemplates